### PR TITLE
Adding DNS container to trace-analytics-sample-app.

### PR DESCRIPTION
**Pull request by wrijeff**
*Issue #, if available:* #200 

*Description of changes:*
Adding a small DNS service to the docker-compose example, to be used in the multi-host DNS discovery example.

Tested with `docker compose up -d` and confirming data still propagates all the way to ES:
>15846 [raw-pipeline-process-worker-3-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker  –  raw-pipeline Worker: Processing 7 records from buffer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/trace-analytics-sample-app/opentelemetry-collector/otel-collector-dns-config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:55680
+
+exporters:
+  otlp/2:
+    endpoint: prepper-cluster:21890
```
Comment (chenqi0805) [inline examples/trace-analytics-sample-app/opentelemetry-collector/otel-collector-dns-config.yml:9]: QUES: just curious. Will 10.10.2.1:21890 also work?

File: examples/dns/dnsmasq.conf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+log-queries
+server=127.0.0.11
```
Comment (chenqi0805) [inline examples/dns/dnsmasq.conf:2]: QUES: Is the IP typo? I actually do not quite understand what this flag means. In the doc it says `Specify IP address of upstream servers directly`

**Review by chenqi0805 — COMMENTED** [review]
Just some of my own questions on this setup. Could you also add some description about how DNS container works in the README?

**Comment by wrijeff** [comment]
> Just some of my own questions on this setup. Could you also add some description about how DNS container works in the README?

Sure, will add documentation.

File: examples/trace-analytics-sample-app/opentelemetry-collector/otel-collector-dns-config.yml (hunk 1 above)
Comment (wrijeff) [reply examples/trace-analytics-sample-app/opentelemetry-collector/otel-collector-dns-config.yml:9]: Correct, any of the following will work.
* `10.10.2.1` - the address of the prepper instance
* `data-prepper` - provided by [docker-compose](https://docs.docker.com/compose/networking/) (hostname = container name)
* `prepper-cluster` - defined in dnsmasq configuration

In this example I'm pointing the otel exporter at `prepper-cluster` so we can easily test with multiple hosts by adding entries to the dnsmasq hosts.prepper file.

File: examples/dns/dnsmasq.conf (hunk 2 above)
Comment (wrijeff) [reply examples/dns/dnsmasq.conf:2]: Not a typo (I can add a comment). 

Docker's internal embedded DNS happens to be at `127.0.0.11`, so by setting this as an upstream server we are telling dnsmasq to relay requests to the Docker's DNS for names which dnsmasq can't resolve.

**Review by chenqi0805 — APPROVED** [review]

**Comment by wrijeff** [comment]
Closing as project structure is changing - will most likely include these changes in a follow-up PR.

**Pull request closed without merging**